paths.go

```diff
@@ -0,0 +1,128 @@
+package quamina
+
+import (
+	"fmt"
+	"strings"
+)
+
+const PathSeparator = "\n"
+
+type pathsIndex struct {
+	root bool
+
+	nodes map[string]*pathsIndex
+
+	// fields map from it's name to it's full path (as specific in quamina.Field)
+	// will be present only on the leafs
+	fields map[string][]byte
+}
+
+func newPathsIndex() *pathsIndex {
+	return &pathsIndex{
+		root:   true,
+		nodes:  make(map[string]*pathsIndex),
+		fields: make(map[string][]byte),
+	}
+}
+
+func (p *pathsIndex) add(path string) {
+	parts := strings.Split(path, PathSeparator)
+
+	if len(parts) == 1 {
+		// It's a direct field.
+		p.fields[path] = []byte(path)
+		return
+	}
+
+	var node *pathsIndex
+
+	node = p.getOrCreate(parts[0])
+
+	for i, part := range parts[1:] {
+		if i == len(parts)-2 {
+			node.addField(part, []byte(path))
+		} else {
+			node = node.getOrCreate(part)
+		}
+	}
+}
+
+func (p *pathsIndex) Get(name []byte) (HierarchalNameTracker, bool) {
+	n, ok := p.nodes[string(name)]
+	return n, ok
+}
+
+func (p *pathsIndex) getOrCreate(name string) *pathsIndex {
+	if _, ok := p.nodes[name]; !ok {
```

Comment:
For consistent style, please break into 2 lines. 
```go
   _, ok := p.nodes[name]
  if !ok { ...
```
(I hate trailing `.   }\n. }\n }`)

(recurs)

Reply:
Change the occurrences in this file.